adapters/silverpush/silverpush.go

```diff
@@ -0,0 +1,344 @@
+		return nil, &errortypes.BadInput{Message: "Invalid or missing video field(s)"}
+	}
+	return &videoCopy, nil
+}
+
+func setBannerDimension(banner *openrtb2.Banner) (*openrtb2.Banner, error) {
+	if banner.W != nil && banner.H != nil {
+		return banner, nil
+	}
+	if len(banner.Format) == 0 {
+		return banner, &errortypes.BadInput{Message: "No sizes provided for Banner."}
+	}
+	bannerCopy := *banner
+	bannerCopy.W = openrtb2.Int64Ptr(banner.Format[0].W)
+	bannerCopy.H = openrtb2.Int64Ptr(banner.Format[0].H)
+
+	return &bannerCopy, nil
+}
+
+func setPublisherId(req *openrtb2.BidRequest, imp *openrtb2.Imp, impExt *openrtb_ext.ImpExtSilverpush) error {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+
+	if err := json.Unmarshal(bidderExt.Bidder, impExt); err != nil {
+		return &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	if impExt.PublisherId == "" {
+		return &errortypes.BadInput{Message: "Missing publisherId parameter."}
+	}
+	if req.Site != nil {
+		siteCopy := *req.Site
+		if siteCopy.Publisher == nil {
+			siteCopy.Publisher = &openrtb2.Publisher{ID: impExt.PublisherId}
+		} else {
+			publisher := *siteCopy.Publisher
+			publisher.ID = impExt.PublisherId
+			siteCopy.Publisher = &publisher
+		}
+		req.Site = &siteCopy
+
+	} else if req.App != nil {
+		appCopy := *req.App
+		if appCopy.Publisher == nil {
+			appCopy.Publisher = &openrtb2.Publisher{ID: impExt.PublisherId}
+		} else {
+			publisher := *appCopy.Publisher
+			publisher.ID = impExt.PublisherId
+			appCopy.Publisher = &publisher
+		}
+		appCopy.Publisher = &openrtb2.Publisher{ID: impExt.PublisherId}
+		req.App = &appCopy
+
+	}
+
+	return nil
+}
+
+func impressionByMediaType(imp *openrtb2.Imp) ([]openrtb2.Imp, error) {
+
+	if imp.Banner == nil && imp.Video == nil {
+		return nil, &errortypes.BadInput{Message: "Invalid MediaType. SilverPush only supports Banner, Video"}
+	}
+	imps := make([]openrtb2.Imp, 0, 2)
+
+	if imp.Banner != nil {
+		impCopy := *imp
+		impCopy.Video = nil
+		imps = append(imps, impCopy)
+	}
+	if imp.Video != nil {
+		impCopy := *imp
+		impCopy.Banner = nil
+		imps = append(imps, impCopy)
+	}
+	return imps, nil
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode)}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode)}}
+	}
```

Comment:
This is correct but you could also use the following code - 
```
if adapters.IsResponseStatusCodeNoContent(response) {
		return nil, nil
	}

	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
		return nil, []error{&errortypes.BadInput{
			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
		}}
	}
```

Reply:
Fixed
[commit](https://github.com/prebid/prebid-server/pull/2957/commits/786baef2e551736d40decb753127e6a90a580b8e)